Fix dectection of fox libs
Allow configuring the locations of lrelease and lupdate
Prevent linking to unneeded Qt5 libraries

--- configure.orig	2017-07-30 14:20:32 UTC
+++ configure
@@ -951,6 +951,8 @@ with_x
 with_qt5_qmake
 with_qt5_moc
 with_qt5_uic
+with_qt5_lrelease
+with_qt5_lupdate
 with_docpath
 enable_release
 enable_memtrace
@@ -1672,6 +1674,8 @@ Optional Packages:
   --with-qt5-qmake=FILE    uses given qmake
   --with-qt5-moc=FILE      uses given qt moc
   --with-qt5-uic=FILE      uses given qt uic
+  --with-qt5-lrelease=FILE     uses given qt lrelease
+  --with-qt5-lupdate=FILE      uses given qt lupdate
   --with-docpath=DIR where to store the apidoc
   --with-buffer-hardlimit=VALUE new default buffer hard limit
   --with-plugins-ct=LIST list of CT plugins to build
@@ -21084,7 +21088,7 @@ $as_echo_n "checking for fox libraries..
 
 # Check whether --with-fox-libs was given.
 if test "${with_fox_libs+set}" = set; then :
-  withval=$with_fox_libs; fox_libs="$withval"
+  withval=$with_fox_libs; local_fox_libs="$withval"
 else
 
       ac_ext=cpp
@@ -22415,43 +22419,16 @@ $as_echo_n "checking for Qt... " >&6; }
     am_have_qt_makefile=`mktemp`
     # http://qt-project.org/doc/qt-5/qmake-variable-reference.html#qt
     cat > $am_have_qt_pro << EOF
-qtHaveModule(axcontainer):       QT += axcontainer
-qtHaveModule(axserver):          QT += axserver
-qtHaveModule(concurrent):        QT += concurrent
 qtHaveModule(core):              QT += core
-qtHaveModule(dbus):              QT += dbus
-qtHaveModule(declarative):       QT += declarative
-qtHaveModule(designer):          QT += designer
 qtHaveModule(gui):               QT += gui
-qtHaveModule(help):              QT += help
-qtHaveModule(multimedia):        QT += multimedia
-qtHaveModule(multimediawidgets): QT += multimediawidgets
-qtHaveModule(network):           QT += network
-qtHaveModule(opengl):            QT += opengl
-qtHaveModule(printsupport):      QT += printsupport
-qtHaveModule(qml):               QT += qml
-qtHaveModule(qmltest):           QT += qmltest
-qtHaveModule(x11extras):         QT += x11extras
-qtHaveModule(script):            QT += script
-qtHaveModule(scripttools):       QT += scripttools
-qtHaveModule(sensors):           QT += sensors
-qtHaveModule(serialport):        QT += serialport
-qtHaveModule(sql):               QT += sql
-qtHaveModule(svg):               QT += svg
-qtHaveModule(testlib):           QT += testlib
-qtHaveModule(uitools):           QT += uitools
-qtHaveModule(webkit):            QT += webkit
-qtHaveModule(webkitwidgets):     QT += webkitwidgets
-qtHaveModule(xml):               QT += xml
-qtHaveModule(xmlpatterns):       QT += xmlpatterns
 qtHaveModule(widgets):           QT += widgets
 percent.target = %
 percent.commands = @echo -n "\$(\$(@))\ "
 QMAKE_EXTRA_TARGETS += percent
 EOF
     $QT_QMAKE $am_have_qt_pro -o $am_have_qt_makefile
-    QT_CXXFLAGS=`make -s -f $am_have_qt_makefile CXXFLAGS INCPATH`
-    QT_LIBS=`make -s -f $am_have_qt_makefile LIBS`
+    QT_CXXFLAGS=`make -s -f $am_have_qt_makefile -VCXXFLAGS -VINCPATH | tr '\n' ' '`
+    QT_LIBS=`make -s -f $am_have_qt_makefile -VLIBS | tr '\n' ' '`
     rm $am_have_qt_pro $am_have_qt_makefile
 
     # Look for specific tools in $PATH
@@ -22474,9 +22451,21 @@ else
 
 fi
 
+# Check whether --with-qt5-lrelease was given.
+if test "${with_qt5_lrelease+set}" = set; then :
+  withval=$with_qt5_lrelease; QT_LRELEASE="$withval"
+else
+  QT_RELEASE=`which lrelease`
+
+fi
+
+# Check whether --with-qt5-lupdate was given.
+if test "${with_qt5_lupdate+set}" = set; then :
+  withval=$with_qt5_lupdate; QT_LUPDATE="$withval"
+else
+  QT_LUPDATE=`which lupdate`
 
-    QT_LRELEASE=`which lrelease`
-    QT_LUPDATE=`which lupdate`
+fi
 
     # Get Qt version from qmake
     QT_DIR=`$QT_QMAKE --version | grep -o -E /.+`
